### PR TITLE
Fix "hide cursor" in eye tracking example scene does not work

### PR DESCRIPTION
**Pull request by sostel**
## Overview
- Workaround for issue #6086 
- Exception handling and removed confusing error messages showing up when switching scenes when using "LoadProfilesOnStartup" (addressing issue #6735).

### CODE REVIEW

File: Assets/MixedRealityToolkit.Examples/Demos/EyeTracking/DemoTargetSelections/Prefabs/TargetType (1).prefab [hunk 1]
```diff
@@ -4895,4 +4895,4 @@ MonoBehaviour:
   m_Name: 
   m_EditorClassIdentifier: 
   RotateByEulerAngles: {x: 0, y: 1, z: 1}
-  speed: 1
+  speed: 0.3
```
Comment (julenka) [inline Assets/MixedRealityToolkit.Examples/Demos/EyeTracking/DemoTargetSelections/Prefabs/TargetType (1).prefab]: Was this change intentional?

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs [hunk 2]
```diff
@@ -361,16 +361,19 @@ protected virtual void Start()
...
+                    if (CoreServices.InputSystem.FocusProvider.IsPointerRegistered(Pointer))
+                    {
+                        Debug.LogError($"{name}: Unable to get focus details for {pointer.GetType().Name}!");
+                    }
                 }
-            }
 
-            UpdateCursorState();
-            UpdateCursorTransform();
+                UpdateCursorState();
```
Comment (julenka) [inline Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs]: Hard to tell from the delta but I think the indentation might be off here.

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InputRecordingControls.cs [hunk 3]
```diff
@@ -20,11 +21,16 @@ private InputRecordingService RecordingService
         {
             get
             {
-                if (recordingService == null)
+                try
```
Comment (julenka) [inline Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InputRecordingControls.cs]: Interesting, could you provide a little context around why you need this Try/Catch here? Perhaps it was unintentionally added?

File: Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs [hunk 4]
```diff
@@ -333,7 +333,7 @@ void IMixedRealitySourceStateHandler.OnSourceLost(SourceStateEventData eventData
                 if (pointer.InputSourceParent == inputSource)
                 {
                     int pointerId = (int)pointer.PointerId;
-                    Debug.Assert(pointerDataToUpdate.ContainsKey(pointerId));
+                   // Debug.Assert(pointerDataToUpdate.ContainsKey(pointerId));
```
Comment (julenka) [inline Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs]: Accidentally commented out code?

File: Assets/MixedRealityToolkit.Tests/EditModeTests/Core/TestFixture_01_MixedRealityToolkitTests.cs [hunk 5]
```diff
@@ -0,0 +1,497 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (julenka) [inline Assets/MixedRealityToolkit.Tests/EditModeTests/Core/TestFixture_01_MixedRealityToolkitTests.cs]: This looks like  a new file that might not need to be added

File: Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs [hunk 6]
```diff
@@ -324,7 +324,7 @@ private void InitializeServiceLocator()
             //If the Mixed Reality Toolkit is not configured, stop.
             if (ActiveProfile == null)
             {
-                Debug.LogError("No Mixed Reality Configuration Profile found, cannot initialize the Mixed Reality Toolkit");
+                // Debug.LogError("No Mixed Reality Configuration Profile found, cannot initialize the Mixed Reality Toolkit");
```
Comment (julenka) [inline Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs]: Accidental comment

File: Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs (hunk 4 above)
Comment (SimonDarksideJ) [reply Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs]: Agreed, release code should not contain comments.  Either remove or (in this case)  revert the change.  Asset statements should be retained

File: Assets/MixedRealityToolkit.Examples/Demos/EyeTracking/DemoTargetSelections/Prefabs/TargetType (1).prefab (hunk 1 above)
Comment (sostel) [reply Assets/MixedRealityToolkit.Examples/Demos/EyeTracking/DemoTargetSelections/Prefabs/TargetType (1).prefab]: The change was intentional, however, I was meaning to submit it in a separate patch. I noticed that the gems in the eye tracking scene rotate VERY was what moving the cursor over them. This is not the expected behavior. Not sure what changed. I can revert it for now until looking into whether the fast behavior is only in the Editor or also on the device.

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs (hunk 2 above)
Comment (sostel) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs]: I think the indentation is correct? I added an if statement at the top, because it was blurting out tons of error messages when loading up scenes after switching profiles.

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InputRecordingControls.cs (hunk 3 above)
Comment (sostel) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InputRecordingControls.cs]: Again related to issue #6735. After switching profiles with the LoadProfilesOnStartup script, I get a number of errors. In this particular case, I assume that CoreServices.InputSystem is briefly null causing an error when trying to access GetDataProvider.

File: Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs (hunk 4 above)
Comment (sostel) [reply Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs]: Related to #6735. Please provide more specific feedback here: Why are you calling a Debug.Assert here? What does this mean to developers? Please provide more descriptive debug information. Otherwise the Editor Console is just filled up with "Assert Failed" errors that are unhelpful and confusing to developers.
image

File: Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs (hunk 6 above)
Comment (sostel) [reply Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs]: Not accidental. Again related to #6735 - my dev console gets filled up with unhelpful and confusing error messages. With switching profiles, it's expected that the ActiveProfile may be briefly null. Hence, showing an error every time the app switches profiles / loads a new scene is not great. How about we show a warning instead?

File: Assets/MixedRealityToolkit.Tests/EditModeTests/Core/TestFixture_01_MixedRealityToolkitTests.cs (hunk 5 above)
Comment (sostel) [reply Assets/MixedRealityToolkit.Tests/EditModeTests/Core/TestFixture_01_MixedRealityToolkitTests.cs]: Oh weird! Good catch!

File: Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs (hunk 6 above)
Comment (sostel) [reply Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs]: Not accidental. Again related to #6735 - my dev console gets filled up with unhelpful and confusing error messages. With switching profiles, it's expected that the ActiveProfile may be briefly null. Hence, showing an error every time the app switches profiles / loads a new scene is not great. How about we show a warning instead?

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs (hunk 2 above)
Comment (sostel) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs]: I think the indentation is correct? I added an if statement at the top checking if the CoreServices.InputSystem.FocusProvider isn't null, because it was blurting out tons of error messages when loading up scenes after switching profiles.
Related to #6735.

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InputRecordingControls.cs (hunk 3 above)
Comment (sostel) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InputRecordingControls.cs]: It's related to issue #6735. After switching profiles with the LoadProfilesOnStartup script, I get a number of errors. In this particular case, I got some NullReferenceException. I assume that CoreServices.InputSystem is briefly null causing an error when trying to access GetDataProvider.

File: Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs (hunk 4 above)
Comment (sostel) [reply Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs]: Related to #6735. Please provide more specific feedback here: Why are you calling a Debug.Assert here? What does this mean to developers? Please provide more descriptive debug information. Otherwise the Editor Console is just filled up with "Assert Failed" errors that are unhelpful and confusing to developers.
image
image
😐

File: Documentation/Architecture/CrossPlatform/DeterminingPlatformCapabilities.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
+# Determining platform capabilities
+
+A common question asked of the MRTK involves knowing which specific device (ex: Microsoft HoloLens 2) is being
+used to run an application. Identifying the exact hardware can be challenging on different platforms. The MRTK
+instead provides a way identify specific capabilities of the system (for example, if the system has the articulated
```
Comment (thalbern) [inline Documentation/Architecture/CrossPlatform/DeterminingPlatformCapabilities.md]:
```suggestion
instead provides a way to identify specific capabilities of the system (for example, if the system has the articulated
```

File: Documentation/Architecture/CrossPlatform/DeterminingPlatformCapabilities.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+hands capability).
+
+## Capabilities
+
+The Mixed Reality Toolkit provides the [MixedRealityCapability](xref:Microsoft.MixedReality.Toolkit.MixedRealityCapability)
+enumeration which defines a set of capabilities for which an application can query at runtime. This enum contains the 
+complete set of capabilities for which the Mixed Reality Toolkit supports checking.
+
+### Input System capabilities
```
Comment (thalbern) [inline Documentation/Architecture/CrossPlatform/DeterminingPlatformCapabilities.md]:
```suggestion
### Input system capabilities
```

File: Documentation/Architecture/CrossPlatform/DeterminingPlatformCapabilities.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+bool supportsArticulatedHands = false;
+
+IMixedRealityCapabilityCheck capabilityCheck = inputSystem as IMixedRealityCapabilityCheck;
+if (capabilityCheck != null)
+{
+    supportsArticulatedHands = capabilityCheck.CheckCapability(MixedRealityCapability.ArticulatedHand);
+}
+```
+
+### Spatial Awareness capabilities
```
Comment (thalbern) [inline Documentation/Architecture/CrossPlatform/DeterminingPlatformCapabilities.md]:
```suggestion
### Spatial awareness capabilities
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 10]
```diff
@@ -1,79 +1,78 @@
-# Configuring Mesh Observers for Editor
+# The Spatial Object Mesh Observer
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
# The spatial object mesh observer
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 11]
```diff
@@ -1,79 +1,78 @@
...
-1. Select the desired *Mixed Reality Toolkit* configuration profile or select the *Mixed Reality Toolkit* object in scene
-1. Open or expand the *Spatial Awareness System* tab
-1. Click on *"Add Spatial Observer"* button
+2. Select the SpatialObjectMeshObserver 
 
-    ![Add Spatial Observer](../../Documentation/Images/SpatialAwareness/AddObserver.png)
+![Select Spatial Object Mesh Observer](../../Documentation/Images/SpatialAwareness/SelectObjectObserver.png)
 
-1. Select the *SpatialObjectMeshObserver* type
+3. Select the desired Spatial Mesh Object
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
3. Select the desired spatial mesh object  
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 12]
```diff
@@ -1,79 +1,78 @@
...
 
-Since the *Spatial Object Mesh Observer* loads data from a 3D model, it does not honor some of the standard mesh
-observer settings which are outlined below.
+The [Windows Device Portal](https://docs.microsoft.com/en-us/windows/mixed-reality/using-the-windows-device-portal) can be used to download the spatial mesh, as a .obj file, from a Microsoft HoloLens device.
 
-**Update Interval**
+1. Scan your environment
+1. Connect to your HoloLens using the Windows Device Portal
+1. Navigate to the 3D View page
+1. Update the Surface reconstruction
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
1. Update the surface reconstruction
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 13]
```diff
@@ -1,79 +1,78 @@
...
 
-1. Select the desired *Spatial Mesh Object*. By default, the observer is configured with an example model. This model was created using a Microsoft HoloLens but it is possible to [create a new scan mesh object](#acquiring-environment-scans).
-1. [Configure rest of the Mesh Observer profile properties](ConfiguringSpatialAwarenessMeshObserver.md)
+By default, the observer is configured with an example model. This model was created using a Microsoft HoloLens.
 
-    ![Select the Mesh Object](../../Documentation/Images/SpatialAwareness/ObjectObserverProfile.png)
+## Acquiring environment scans
 
-### Spatial Object Mesh Observer profile notes
+### Windows Device Portal
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
### Windows device portal
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 14]
```diff
@@ -1,79 +1,78 @@
...
 
-Since the *Spatial Object Mesh Observer* loads data from a 3D model, it does not honor some of the standard mesh
-observer settings which are outlined below.
+The [Windows Device Portal](https://docs.microsoft.com/en-us/windows/mixed-reality/using-the-windows-device-portal) can be used to download the spatial mesh, as a .obj file, from a Microsoft HoloLens device.
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
The [Windows device portal](https://docs.microsoft.com/en-us/windows/mixed-reality/using-the-windows-device-portal) can be used to download the spatial mesh, as a .obj file, from a Microsoft HoloLens device.
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 15]
```diff
@@ -1,79 +1,78 @@
...
 
-**Update Interval**
+1. Scan your environment
+1. Connect to your HoloLens using the Windows Device Portal
+1. Navigate to the 3D View page
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
1. Navigate to the 3D view page
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 16]
```diff
@@ -1,79 +1,78 @@
...
-### Spatial Object Mesh Observer profile notes
+### Windows Device Portal
 
-Since the *Spatial Object Mesh Observer* loads data from a 3D model, it does not honor some of the standard mesh
-observer settings which are outlined below.
+The [Windows Device Portal](https://docs.microsoft.com/en-us/windows/mixed-reality/using-the-windows-device-portal) can be used to download the spatial mesh, as a .obj file, from a Microsoft HoloLens device.
 
-**Update Interval**
+1. Scan your environment
+1. Connect to your HoloLens using the Windows Device Portal
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
1. Connect to your HoloLens using the Windows device portal
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 17]
```diff
@@ -1,79 +1,78 @@
...
+1. Save the file to your PC
 
-The  *Spatial Object Mesh Observer* sends all meshes to an application when the model is loaded. It does not
-simulate time deltas between updates. An application can re-receive the mesh events by calling [`myObserver.ClearObservation()`](xref:Microsoft.MixedReality.Toolkit.SpatialAwareness.IMixedRealitySpatialAwarenessObserver.ClearObservations) and [`myObserver.Resume()`](xref:Microsoft.MixedReality.Toolkit.SpatialAwareness.IMixedRealitySpatialAwarenessObserver.Resume).
+### HoloToolkit .room files
 
-**Is Stationary Observer**
+Many developers will have previously used HoloToolkit to scan environments and create .room files. The 
+Mixed Reality Toolkit now supports importing these files as GameObjects in Unity and use them as
+Spatial Mesh Objects in the observer.
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
Spatial mesh objects in the observer.
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 18]
```diff
@@ -1,79 +1,78 @@
...
 
-The *Spatial Object Mesh Observer* considers all 3D mesh objects to be stationary and disregards origin.
+## Observer behavior notes
 
-**Observer Shape and Extents**
+Since the spatial object mesh observer loads data from a 3D model, it does not honor some of the standard mesh
+observer settings. The following sections describe these settings.
 
-The  *Spatial Object Mesh Observer* sends the entire 3D mesh to the application. Observer shape and extents are not considered.
+### Update Interval
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
### Update interval
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 19]
```diff
@@ -1,79 +1,78 @@
...
+An application can re-receive the mesh events by calling ClearObservations() and the Resume(). 
 
-## Acquiring environment scans
+### Is Stationary Observer
 
-This section outlines additional information to create and gather *Spatial Mesh Object* files for use with the *Spatial Object Mesh Observer*.
+The spatial observer considers all 3D mesh objects to be stationary.
 
-### Windows Device Portal
+### Observer Shape and Extents
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
### Observer shape and extents
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 20]
```diff
@@ -1,79 +1,78 @@
...
 
-The [Windows Device Portal](https://docs.microsoft.com/en-us/windows/mixed-reality/using-the-windows-device-portal) can be used to download the spatial mesh, as a .obj file, from a Microsoft HoloLens device.
+The spatial object meshobserver sends the entire 3D mesh to the application. Observer shape and extents are not considered.
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
The spatial object mesh observer sends the entire 3D mesh to the application. Observer shape and extents are not considered.
```

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md [hunk 21]
```diff
@@ -1,79 +1,78 @@
...
 
-1. Scan by simply walking and viewing the desired environment with a HoloLens
-1. Connect to the HoloLens using the Windows Device Portal
-1. Navigate to the *3D View* page
-1. Click the *Update* button under *Spatial Mapping* section
-1. Click the *Save* button under *Spatial Mapping* section to save the obj file to PC
+## Level of Detail and Triangles / Cubic Meter
```
Comment (thalbern) [inline Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]:
```suggestion
## Level of detail and triangles / cubic meter
```

File: Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/SpeechInputHandler.cs [hunk 22]
```diff
@@ -47,6 +47,7 @@ public SpeechConfirmationTooltip SpeechConfirmationTooltipPrefab
         private readonly Dictionary<string, UnityEvent> responses = new Dictionary<string, UnityEvent>();
 
         #region MonoBehaviour Implementation
+        private bool rememberFocusRequired;
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/SpeechInputHandler.cs]: this seems a bit hacky to me - what magic is happening when manually focusing the speech input handler on start? is there a better way to solve this? and if we have to go with the hack can we comment why we're doing this?

File: Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs (hunk 4 above)
Comment (thalbern) [reply Assets/MixedRealityToolkit.Services/InputSystem/MixedRealityInputModule.cs]: i'm not sure this needs to be an assertion? it could also be a warning imho.
also i would give more information to the user if we decide to keep this - the current assert log is not providing any information. maybe like: "Trying to remove a pointer in that wasn't registered".
same goes for assert a few lines below: "Trying to remove a pointer that has already been removed"

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InputRecordingControls.cs (hunk 3 above)
Comment (keveleigh) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InputRecordingControls.cs]: Can we do a null check for `CoreServices.InputSystem` instead of a `Try/Catch`?

File: Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs [hunk 23]
```diff
@@ -324,7 +324,7 @@ private void InitializeServiceLocator()
             //If the Mixed Reality Toolkit is not configured, stop.
             if (ActiveProfile == null)
             {
-                Debug.LogError("No Mixed Reality Configuration Profile found, cannot initialize the Mixed Reality Toolkit");
+                Debug.LogWarning("No Mixed Reality Configuration Profile found, cannot initialize the Mixed Reality Toolkit");
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs]: I think this should stay as an error. Having no active profile is not a state we should ever be in.

```suggestion
                Debug.LogError("No Mixed Reality Configuration Profile found, cannot initialize the Mixed Reality Toolkit");
```

Comment (keveleigh) [reply Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs]: If this is happening because we're manually setting the profile to null for some reason, it should just be swapped directly to whichever new profile is going to be set.

If we see issues with that approach, pinging @davidkline-ms  :)

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md (hunk 10 above)
Comment (keveleigh) [reply Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]: @thalbern Should headers / titles for the entire page stay in caps?

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md (hunk 14 above)
Comment (keveleigh) [reply Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]: I think Windows Device Portal is considered branding maybe and is capitalized as-is in our platform docs:

image

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md (hunk 10 above)
Comment (thalbern) [reply Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]: no, they should be lower case - it's ms docs styleguide - don't do random capitalization  - only names should be uppercase

File: Documentation/SpatialAwareness/SpatialObjectMeshObserver.md (hunk 14 above)
Comment (thalbern) [reply Documentation/SpatialAwareness/SpatialObjectMeshObserver.md]: i wasn't sure about that one - if official ms docs are putting it in uppercase we have to as well :) thanks for checking kurtis!

File: Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs (hunk 23 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs]: I tend to agree, @keveleigh

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs [hunk 24]
```diff
@@ -361,16 +361,19 @@ protected virtual void Start()
 
         private void Update()
         {
-            if (!CoreServices.InputSystem.FocusProvider.TryGetFocusDetails(Pointer, out focusDetails))
+            if ((CoreServices.InputSystem != null) && (CoreServices.InputSystem.FocusProvider != null))
```
Comment (julenka) [inline Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Cursors/BaseCursor.cs]: I think this may no longer be needed as I think @davidkline-ms  has fixed issues regarding switching mrtk profiles at runtime.

File: Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InputRecordingControls.cs (hunk 3 above)
Comment (julenka) [reply Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/InputRecordingControls.cs]: It seems like this should be fixed by #7275 once that is in, so perhaps not need to add the workaround?

File: Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/SpeechInputHandler.cs [hunk 25]
```diff
@@ -75,6 +76,9 @@ protected override void Start()
                     responses.Add(keyword, keywordAndResponse.Response);
                 }
             }
+
+            rememberFocusRequired = this.IsFocusRequired;
```
Comment (julenka) [inline Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/SpeechInputHandler.cs]: Could you explain why you are needing to save off IsFocusRequired? Is it beause of how the eye tracking example scenes are registering for speech commands? My understanding is "hide cursor" would be a global speech command handled by a global handler so I'm confused why we need to do this save / restore here.

**Comment by julenka** [comment]
Hi @sostel thank you for working on the eye tracking scene fixes! I'm attempting to revive this PR as it's been out for a few months, so I added a few comments.

**Comment by wiwei** [comment]
Closing out some old PRs - feel free to reopen when folks have a chance to work on this again.
